war/src/main/java/io/lumeer/engine/controller/SecurityFacade.java

```diff
@@ -51,9 +52,377 @@
    private DataStorage dataStorage;
 
    @Inject
+   @SystemDataStorage
+   private DataStorage systemDataStorage;
+
+   @Inject
    private DataStorageDialect dataStorageDialect;
 
    @Inject
+   private UserFacade userFacade;
+
+   /********* CHECKING ROLES *********/
+
+   public boolean hasOrganizationRole(String organizationId, String role) {
+      DataDocument doc = systemDataStorage.readDocumentIncludeAttrs(
+            LumeerConst.Security.ORGANIZATION_ROLES_COLLECTION_NAME,
+            dataStorageDialect.fieldValueFilter(LumeerConst.Security.ORGANIZATION_ID_KEY, organizationId),
+            Arrays.asList(LumeerConst.Security.ROLES_KEY));
+
+      return checkRole(doc, userFacade.getUserEmail(), role);
+   }
+
+   public boolean hasProjectRole(String projectId, String role) {
+      DataDocument doc = dataStorage.readDocumentIncludeAttrs(
+            LumeerConst.Security.ROLES_COLLECTION_NAME,
+            dataStorageDialect.multipleFieldsValueFilter(projectFilter(projectId)),
+            Arrays.asList(dataStorageDialect.concatFields(LumeerConst.Security.ROLES_KEY, role)));
+
+      return checkRole(doc, userFacade.getUserEmail(), role);
+   }
+
+   public boolean hasCollectionRole(String collectionName, String projectId, String role) {
+      DataDocument doc = dataStorage.readDocumentIncludeAttrs(
+            LumeerConst.Security.ROLES_COLLECTION_NAME,
+            dataStorageDialect.multipleFieldsValueFilter(collectionFilter(collectionName, projectId)),
+            Arrays.asList(dataStorageDialect.concatFields(LumeerConst.Security.ROLES_KEY, role)));
+
+      return checkRole(doc, userFacade.getUserEmail(), role);
+   }
+
+   public boolean hasViewRole(int viewId, String projectId, String role) {
+      DataDocument doc = dataStorage.readDocumentIncludeAttrs(
+            LumeerConst.Security.ROLES_COLLECTION_NAME,
+            dataStorageDialect.multipleFieldsValueFilter(viewFilter(viewId, projectId)),
+            Arrays.asList(dataStorageDialect.concatFields(LumeerConst.Security.ROLES_KEY, role)));
+
+      return checkRole(doc, userFacade.getUserEmail(), role);
+   }
+
+   private boolean checkRole(DataDocument rolesDocument, String user, String role) {
+      List<String> groupsForUser = Collections.emptyList(); // TODO: user UsersGroupFacade
+      String key = dataStorageDialect.concatFields(LumeerConst.Security.ROLES_KEY, role);
+
+      // does not work :-(
```

Comment:
I don't know if it is necessary to do something right now, as the workaround works, but I told Kubo about it.